models/classes/Lists/DataAccess/Repository/DependentPropertiesRepository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace oat\tao\model\Lists\DataAccess\Repository;
+
+use core_kernel_classes_Property;
+use oat\generis\model\GenerisRdf;
+use oat\generis\model\OntologyRdf;
+use oat\oatbox\service\ConfigurableService;
+use oat\search\helper\SupportedOperatorHelper;
+use oat\tao\model\Lists\Business\Domain\DependentPropertiesRepositoryContext;
+use oat\generis\model\kernel\persistence\smoothsql\search\ComplexSearchService;
+use oat\tao\model\Lists\Business\Contract\DependentPropertiesRepositoryInterface;
+
+class DependentPropertiesRepository extends ConfigurableService implements DependentPropertiesRepositoryInterface
+{
+    /**
+     * @return core_kernel_classes_Property[]
+     */
+    public function findAll(DependentPropertiesRepositoryContext $context): array
+    {
+        $dependentProperties = [];
+        /** @var core_kernel_classes_Property $property */
+        $property = $context->getParameter(DependentPropertiesRepositoryContext::PARAM_PROPERTY);
+
+        $search = $this->getComplexSearchService();
+        $dependentPropertiesQueryBuilder = $search->query();
+        $dependentPropertiesQuery = $search->searchType(
+            $dependentPropertiesQueryBuilder,
+            OntologyRdf::RDF_PROPERTY,
+            true
+        );
+        $dependentPropertiesQuery->addCriterion(
+            GenerisRdf::PROPERTY_DEPENDS_ON_PROPERTY,
+            SupportedOperatorHelper::EQUAL,
+            $property->getUri()
+        );
+        $dependentPropertiesQueryBuilder->setCriteria($dependentPropertiesQuery);
+        $result = $search->getGateway()->search($dependentPropertiesQueryBuilder);
+
+        /** @var string $dependentPropertyUri */
+        foreach ($result as $dependentProperty) {
+            $dependentProperties[] = $dependentProperty;
+        }
+
+        return $dependentProperties;
```

Comment:
You can return it directly, no?

```suggestion
        return $search->getGateway()->search($dependentPropertiesQueryBuilder);
```

Reply:
![image](https://user-images.githubusercontent.com/59471572/134345675-bccb7046-c6c8-4ed5-b525-b561ae19188b.png)

Reply:
You can cast to array

Reply:
I will check🙂